planner/core/exhaust_physical_plans.go

```diff
@@ -2626,11 +2626,30 @@ func (la *LogicalAggregation) exhaustPhysicalPlans(prop *property.PhysicalProper
 }
 
 func (p *LogicalSelection) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]PhysicalPlan, bool, error) {
+	newProps := make([]*property.PhysicalProperty, 0, 3)
 	childProp := prop.CloneEssentialFields()
-	sel := PhysicalSelection{
-		Conditions: p.Conditions,
-	}.Init(p.ctx, p.stats.ScaleByExpectCnt(prop.ExpectedCnt), p.blockOffset, childProp)
-	return []PhysicalPlan{sel}, true, nil
+	newProps = append(newProps, childProp)
+	if p.canPushToCop(kv.TiFlash) &&
+		expression.CanExprsPushDown(p.SCtx().GetSessionVars().StmtCtx, p.Conditions, p.SCtx().GetClient(), kv.TiFlash) {
+		if prop.TaskTp != property.CopSingleReadTaskType {
```

Comment:
If `prop.TaskTP == property.MppTaskType`, then is it neccessary to add `CopSingleReadTaskType`?

Reply:
updated, mpp || cop tiflash, tikv